src/storage/documentService/documentService.js

```diff
@@ -47,19 +68,35 @@ const validateTtl = ttl => {
 
 const uploadDocument = async (
   document,
+  documentId,
   ttl = DEFAULT_TTL,
   network = config.network
 ) => {
   const verificationResults = await verify(document, network);
-  if (!verificationResults.valid) throw new Error("Document is not valid");
+  if (
+    (!documentId && !verificationResults.valid) ||
+    (documentId && !verificationResults.hash.checksumMatch)
+  ) {
+    throw new Error("Document is not valid");
+  }
+
   validateTtl(ttl);
+
+  const placeHolderObj = documentId
+    ? await getPlaceholderObject(documentId)
```

Comment:
i think it might be a lot easier to understand if the function is named getDecryptionKey

Reply:
changed